rust/src/pgsql/pgsql.rs

```diff
@@ -341,16 +356,27 @@ impl PgsqlState {
             );
             match PgsqlState::state_based_req_parsing(self.state_progress, start) {
                 Ok((rem, request)) => {
-                    sc_app_layer_parser_trigger_raw_stream_reassembly(flow, Direction::ToServer as i32);
                     start = rem;
-                    if let Some(state) = PgsqlState::request_next_state(&request) {
+                    let temp_state: PgsqlStateProgress;
+                    let tx_completed = if let Some(state) = PgsqlState::request_next_state(&request) {
```

Comment:
I think the code is much cleaner now, thank. Do feel it could use a comment explaining the state transition work flow